lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2885,10 +2895,21 @@ void TFDeabstractionPass::run() {
   // iff they look like they could be the top level of a deabstraction
   // context.
   for (auto &fn : *module) {
+    // There's no point in deabstracting things defined in other modules,
+    // because we won't lower them.
+    if (fn.isAvailableExternally()) {
+      fn.skippedByDeabstraction = true;
+      continue;
+    }
+
     // If this function is a building block of larger tensor programs (e.g.
     // the ops defined in the TensorFlow module), then don't transform it in
     // isolation.
-    if (!tfc.shouldBePartitioned(&fn, /*forceTFFunctions*/false)) {
+    // However, in dynamic compilation mode, deabstract everything in this
```

Comment:
IIUC, "deabstract everything in this module" is just a short-term plan for eager mode, and the real reason there is because we need to extend/fix irgen before we can turn off deabstraction entirely, correct? consider adding a TODO.

Reply:
will do